src/Persistence/Mapping/Driver/ColocatedMappingDriver.php

```diff
@@ -203,14 +181,17 @@ public function getAllClassNames(): array
     }
 
     /**
-     * @param iterable<SplFileInfo> $filesIterator
+     * @param iterable<TKey, T> $iterable1
+     * @param iterable<TKey, T> $iterable2
+     *
+     * @return iterable<TKey, T>
      *
-     * @return Generator<int,string>
+     * @template TKey
+     * @template T
      */
-    private function pathNameIterator(iterable $filesIterator): Generator
```

Comment:
Technically, removing a private method from a trait is a breaking change. It could be used by the class that uses it in an other package. But this code has not been released yet.

That's why I thing it's better not introducing `concatIterables` in this trait.